src/software/simulated_tests/simulated_tactic_test_fixture.cpp

```diff
@@ -5,85 +5,181 @@
         RobotNavigationObstacleFactory(
-            thunderbots_config->getRobotNavigationObstacleConfig()),
-        thunderbots_config->getNavigatorConfig());
+            friendly_thunderbots_config->getRobotNavigationObstacleConfig()),
+        friendly_thunderbots_config->getNavigatorConfig());
+    enemy_navigator = std::make_shared<Navigator>(
+        std::make_unique<VelocityObstaclePathManager>(
+            std::make_unique<ThetaStarPathPlanner>(),
+            RobotNavigationObstacleFactory(
+                enemy_thunderbots_config->getRobotNavigationObstacleConfig())),
+        RobotNavigationObstacleFactory(
+            enemy_thunderbots_config->getRobotNavigationObstacleConfig()),
+        enemy_thunderbots_config->getNavigatorConfig());
 }
 
-void SimulatedTacticTestFixture::setTactic(std::shared_ptr<Tactic> tactic)
+void SimulatedTacticTestFixture::setTactic(std::shared_ptr<Tactic> friendly_tactic,
+                                           std::shared_ptr<Tactic> enemy_tactic)
+{
+    try
+    {
+        setFriendlyTactic(friendly_tactic);
+        setEnemyTactic(enemy_tactic);
+    }
+    catch (std::invalid_argument& e)
+    {
+        throw(e);
+    }
+}
+
+void SimulatedTacticTestFixture::setFriendlyTactic(std::shared_ptr<Tactic> tactic)
 {
     if (tactic)
     {
-        this->tactic = tactic;
+        this->friendly_tactic = tactic;
     }
     else
     {
-        throw std::invalid_argument("Tactic is invalid");
+        throw std::invalid_argument("Friendly tactic is invalid");
     }
 }
 
-void SimulatedTacticTestFixture::setRobotId(RobotId robot_id)
+void SimulatedTacticTestFixture::setEnemyTactic(std::shared_ptr<Tactic> tactic)
 {
-    this->robot_id = robot_id;
+    if (tactic)
+    {
+        this->enemy_tactic = tactic;
+    }
+    else
+    {
+        throw std::invalid_argument("Enemy tactic is invalid");
+    }
+}
+
+void SimulatedTacticTestFixture::setRobotId(RobotId friendly_robot_id,
+                                            RobotId enemy_robot_id)
+{
+    this->friendly_robot_id = friendly_robot_id;
+    if (enemy_robot_id < UINT_MAX)
+    {
+        this->enemy_robot_id = enemy_robot_id;
+    }
 }
 
 void SimulatedTacticTestFixture::setMotionConstraints(
-    const std::set<MotionConstraint>& motion_constraints)
+    const std::set<MotionConstraint>& friendly_motion_constraints,
+    const std::set<MotionConstraint>& enemy_motion_constraints)
 {
-    this->motion_constraints = motion_constraints;
+    this->friendly_motion_constraints = friendly_motion_constraints;
+    this->enemy_motion_constraints    = enemy_motion_constraints;
 }
 
 void SimulatedTacticTestFixture::updatePrimitives(
+    const World& friendly_world, const World& enemy_world,
+    std::shared_ptr<Simulator> simulator_to_update)
+{
+    updateFriendlyPrimitives(friendly_world, simulator_to_update);
+    if (this->enemy_robot_id)
+    {
+        updateEnemyPrimitives(enemy_world, simulator_to_update);
+    }
```

Comment:
doesn't updateEnemyPrimitives check that enemy_robot_id exists?

Reply:
it log fatal's if we cant find an enemy robot id. The idea is to not call the function if we don't need to. `updateEnemyPrimitives` should only get called if there are enemy robots to update. 

@xyztiger I might not be remembering it correctly, but this is how I understand it